plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/DigitalExchangeComponentInstallationServiceImpl.java

```diff
@@ -34,73 +36,107 @@
 
-        ComponentInstallationJob job = createNewJob(digitalExchange, componentId);
+        DigitalExchangeJob job = createNewJob(digitalExchange, componentId, JobType.UNINSTALL);
         job.setUser(username);
         dao.createComponentInstallationJob(job);
 
+        this.executeJob(job);
+
+        return job;
+    }
+
+
+    private void executeJob(DigitalExchangeJob job) {
         CompletableFuture.runAsync(() -> {
             try {
-                componentInstaller.install(job, dao::updateComponentInstallationJob);
+                switch (job.getJobType()) {
+                    case UNINSTALL:
+                        digitalExchangeJobExecutor.uninstall(job, dao::updateComponentInstallationJob);
+                        break;
+                    case INSTALL:
+                        digitalExchangeJobExecutor.install(job, dao::updateComponentInstallationJob);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Not supported job type " + job.getJobType().name()) ;
+                }
             } catch (Throwable ex) {
-                logger.error("Error while installing " + componentId, ex);
-                job.setStatus(InstallationStatus.ERROR);
+                logger.error("Error while executing job for " + job.getComponentId(), ex);
+                job.setStatus(JobStatus.ERROR);
                 job.setErrorMessage(ex.getMessage());
                 job.setEnded(new Date());
                 dao.updateComponentInstallationJob(job);
             }
         });
-
-        return job;
     }
 
-    private synchronized void checkIfAlreadyRunning(String componentId) {
 
-        dao.findLast(componentId).ifPresent(job -> {
-            if (job.getStatus() != InstallationStatus.COMPLETED
-                    && job.getStatus() != InstallationStatus.ERROR) {
+    private synchronized void checkIfAlreadyRunning(String componentId, JobType jobType) {
+
+        dao.findLast(componentId, jobType).ifPresent(job -> {
+            if (job.getStatus() != JobStatus.COMPLETED
+                    && job.getStatus() != JobStatus.ERROR) {
                 BeanPropertyBindingResult bindingResult = new BeanPropertyBindingResult(componentId, "component");
                 bindingResult.reject(ERRCODE_COMPONENT_INSTALLATION_RUNNING, new String[]{}, "digitalExchange.installation.alreadyRunning");
                 throw new ValidationConflictException(bindingResult);
             }
         });
     }
 
-    private ComponentInstallationJob createNewJob(DigitalExchange digitalExchange, String componentId) {
+    private DigitalExchangeJob createNewJob(DigitalExchange digitalExchange, String componentId, JobType type) {
 
-        ComponentInstallationJob job = new ComponentInstallationJob();
+        DigitalExchangeJob job = new DigitalExchangeJob();
+        //TODO let the database create this natively
```

Comment:
This would be nice, but we have to check if it could lead to compatibility issues, since we are supporting multiple RDBMS.